lib/apiservers/engine/backends/container.go

```diff
@@ -824,6 +826,37 @@ func (c *Container) imageExist(imageID string) (storeName string, err error) {
 	return host, nil
 }
 
+func IsContainerRunning(containerID string) (bool, error) {
+	//FIXME:  Instead of using container list path, switch to container inspect
+	// path once the port layer side is implemented.
+
+	// Get an API client to the portlayer
+	portLayerClient := PortLayerClient()
+	if portLayerClient == nil {
+		return false, derr.NewErrorWithStatusCode(fmt.Errorf("Failed to get Port layer client"),
+			http.StatusInternalServerError)
+	}
+
+	all := new(bool)
+	*all = true
+	containList, err := portLayerClient.Containers.GetContainerList(containers.NewGetContainerListParams().WithAll(all))
+	if err != nil {
+		return false, derr.NewErrorWithStatusCode(fmt.Errorf("Failed to get container list: %s", err), http.StatusInternalServerError)
+	}
+
+	for _, t := range containList.Payload {
```

Comment:
This is not going to scale with number of containers at all, this is O(n) for no good reason. Why are we asking all and then iterating over them one by one? Can't we ask the individual container to portlayer? If not sounds like a design problem to me...